lib/runtime/extrinsic/unchecked_extrinsic.go

```diff
@@ -0,0 +1,236 @@
+type Signature struct {
+	Address []byte
+	Sig     []byte
+	Extra   []byte
+}
+
+// UncheckedExtrinsic generic implementation of pre-verification extrinsic
+type UncheckedExtrinsic struct {
+	Signature Signature
+	Function  Function
+}
+
+// CreateUncheckedExtrinsic builds UncheckedExtrinsic given function interface, index, genesisHash and Keypair
+func CreateUncheckedExtrinsic(fnc *Function, index *big.Int, genesisHash common.Hash, signer crypto.Keypair) (*UncheckedExtrinsic, error) {
+	extra := struct {
+		Era                      [1]byte // TODO determine how Era is determined (Immortal is [1]byte{0}, Mortal is [2]byte{X, 0}, Need to determine how X is calculated)
+		Nonce                    *big.Int
+		ChargeTransactionPayment *big.Int
+	}{
+		[1]byte{0},
+		index,
+		big.NewInt(0),
+	}
+	additional := struct {
+		SpecVersion uint32
+		//TransactionVersion uint32
+		GenesisHash  common.Hash
+		GenesisHash2 common.Hash
+	}{specVersion, genesisHash, genesisHash}
+
+	rawPayload := fromRaw(fnc, extra, additional)
+	rawEnc, err := rawPayload.Encode()
+	if err != nil {
+		return nil, err
+	}
+
+	sig, err := signer.Sign(rawEnc)
+	if err != nil {
+		return nil, err
+	}
+
+	extraEnc, err := scale.Encode(extra)
+	if err != nil {
+		return nil, err
+	}
+	// TODO this changes mortality, determine how to set
+	//extraEnc = append([]byte{22}, extraEnc...)
+
+	signature := Signature{
+		Address: signer.Public().Encode(),
+		Sig:     sig,
+		Extra:   extraEnc,
+	}
+	ux := &UncheckedExtrinsic{
+		Function:  *fnc,
+		Signature: signature,
+	}
+	return ux, nil
+}
+
+// CreateUncheckedExtrinsicUnsigned to build unsigned extrinsic
+func CreateUncheckedExtrinsicUnsigned(fnc *Function) (*UncheckedExtrinsic, error) {
+	ux := &UncheckedExtrinsic{
+		Function: *fnc,
+	}
+	return ux, nil
+}
+
+// Encode scale encode UncheckedExtrinsic
+func (ux *UncheckedExtrinsic) Encode() ([]byte, error) {
+	// TODO deal with signed/unsigned encoding
+	enc := []byte{}
+	sigEnc, err := ux.Signature.Encode()
+	if err != nil {
+		return nil, err
+	}
+	enc = append(enc, sigEnc...)
+
+	fncEnc, err := ux.Function.Encode()
+	if err != nil {
+		return nil, err
+	}
+	enc = append(enc, fncEnc...)
+
+	sEnc, err := scale.Encode(enc)
+	if err != nil {
+		return nil, err
+	}
+
+	return sEnc, nil
+}
+
+// Encode to encode Signature type
+func (s *Signature) Encode() ([]byte, error) {
+	enc := []byte{}
+	//TODO determine why this 255 byte is here
+	addEnc, err := scale.Encode(append([]byte{255}, s.Address...))
+	if err != nil {
+		return nil, err
+	}
+	enc = append(enc, addEnc...)
+	// TODO find better way to handle keytype
+	enc = append(enc, []byte{1}...) //this seems to represent signing key type 0 - Ed25519, 1 - Sr22219, 2 - Ecdsa
+	enc = append(enc, s.Sig...)
+	enc = append(enc, s.Extra...)
+	return enc, nil
+}
+
+// Encode scale encode the UncheckedExtrinsic
+func (f *Function) Encode() ([]byte, error) {
+	enc := []byte{byte(f.Call), byte(f.Pallet)}
```

Comment:
f.Pallet is `byte` here but in the type declaration it's an `int`? maybe change the type to `byte`? (line 47)